### ISSUE TITLE
Provide signed builds

### ISSUE THREAD

**Issue by pardom** [facebook/flipper#1457]
## 🐛 Bug Report

The security policy at my company requires apps to be signed and we, therefore, cannot use Flipper.

![security](https://user-images.githubusercontent.com/33068588/89792179-2ff8d980-daf2-11ea-9e05-bc31b731c0dc.png)

Could you add signing to your release process?

**Comment by passy** [comment]
Sorry, at the moment we don't have a way of doing that as the build process happens as part of a GitHub Action in public.

**Comment by pardo-stripe** [comment]
I'm not very familiar with app signing, but I've used GitHub secrets to sign Maven artifacts for publishing. I did a quick search for app signing with GitHub actions and found this:

https://medium.com/better-programming/indie-mac-app-devops-with-github-actions-b16764a3ebe7
https://github.com/bacongravy/mac-app-devops-demo

Perhaps you've already explored these actions and found them not to work. Can you share the efforts you've made toward app signing so the community can offer assistance? I'd be glad to contribute so that I can continue using Flipper daily at work.

**Comment by pardom** [comment]
I'm not very familiar with app signing, but I've used GitHub secrets to sign Maven artifacts for publishing. I did a quick search for app signing with GitHub actions and found this:

https://medium.com/better-programming/indie-mac-app-devops-with-github-actions-b16764a3ebe7
https://github.com/bacongravy/mac-app-devops-demo

Perhaps you've already explored these actions and found them not to work. Can you share the efforts you've made toward app signing so the community can offer assistance? I'd be glad to contribute so that I can continue using Flipper daily at work.

**Comment by ColtonIdle** [comment]
@pardom what happens if you right click the file and press open. Does it give you the option to "open and accept security risk" that way?

**Comment by pardom** [comment]
@ColtonIdle No, that option is not available to me.

**Comment by hsavit1** [comment]
I agree - not having signed builds is a significant impedance to upgrading

The Flipper team needs to figure out the upgrade process if it wants everyone involved to contribute to building a better product

**Comment by renchap** [comment]
With Github environments you can now have signing secrets stored in Github and only readable for the production release workflow, which can only be launched by an approved person (so they can check all is legit and nobody changed the workflow to leak those secrets): https://docs.github.com/en/actions/reference/environments

**Comment by henryzx** [comment]
Noticed that current flipper upgrade pace become more and more frequent.
It becomes a problem now, for that I have to do a manual routine to move flipper out from quarantine each time.
Really looking forward to solve this problem. @passy

**Comment by aforty** [comment]
I've honestly learned to just ignore Flipper updates because of how painful it is to upgrade an app that isn't signed.

**Comment by passy** [comment]
Sorry for making this so difficult for you, folks. We don't have a good solution for this right now. We will likely pick this up in Q3/Q4 depending on where Apple is with M1 MBPs which only run signed code (without Rosetta 2). Right now, internally we cannot sign our code at all because it breaks incremental updates.

If someone wants to look into the steps that would be necessary to sign as part of a GitHub Action though, I'd be happy to take a look at any PRs on that front.

**Comment by andreialecu** [comment]
There are full instructions on how this could be done as part of github actions here if anyone wants to try implementing it: https://localazy.com/blog/how-to-automatically-sign-macos-apps-using-github-actions

Looks pretty simple.

**Comment by troy-lamerton** [comment]
I would like to share Flipper with the rest of my team. This is blocking me because there's no way I can give something that encourages you to trash it on first launch.

See screenshot: 
![Screen Shot 2021-07-28 at 07 26 24](https://user-images.githubusercontent.com/13446008/127268612-cb9c006c-d3df-4897-9521-9196bdbc8eb0.png)

**Comment by thevoiceless** [comment]
> We will likely pick this up in Q3/Q4 depending on where Apple is with M1 MBPs 

Any updates on this issue?

**Comment by micnguyen** [comment]
For those that are hard blocked and need a path forward running Flipper on your managed Mac, you can run `xattr -d com.apple.quarantine /Applications/Flipper.app` which will work. It's not recommended and may ding you with your organisation's security team so use it with caution.

**Comment by archcorsair** [comment]
I would like to bump this while also expressing interest in signed builds, even when installing via homebrew I am forced to go through the dance of opening it multiple times to ignore the unsigned build.

**Comment by archcorsair** [comment]
@thevoiceless Awesome, thanks! I've added the script to my update flow, hopefully it works for the next update.

**Comment by ponchautf** [comment]
any progress on this?

It become painful as Flipper get updated almost every weeks.
Within a large team with corporate mac (with a lot of security setting that prevent us to bypass signed witjout having to open issue to our support team) is not managable at all

We are thinking to definitively quit flipper to the old Chrome console that  suit for most workload
